### PR TITLE
Add support for .azuredevops folder

### LINKED ISSUE #4034: [Feature request] Allow .azuredevops as well as .azdo folders
Currently, `azd` looks for the [azure-dev.yml](https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/configure-devops-pipeline?tabs=GitHub#prerequisites) file in either:

- For GitHub Actions: in the `.github/workflows` directory.
- For Azure DevOps: in the `.azdo/pipelines` directory.

The ability to use [Azure DevOps pull request templates](https://learn.microsoft.com/en-us/azure/devops/repos/git/pull-request-templates?view=azure-devops) supports placing a template in the following directories - `.azuredevops`, `.vsts`, `docs` or the root of the repository.

As a result, we've implemented the convention of using `.azuredevops` as the directory where all our azdo components live. It would be great to be able to use either `.azdo` or `.azuredevops` directories with `azd`.

My proposal is to allow `azd` to be able to look for `azure-dev.yml` in either `.azdo/pipelines/azure-dev.yml` or `.azuredevops/pipelines/azure-dev.yml` folders when using Azure DevOps.

### PR DESCRIPTION
**Pull request by jongio**
Fixes #4034

### CODE REVIEW

File: .vscode/cspell.global.yaml [hunk 1]
```diff
@@ -96,6 +96,7 @@ ignoreWords:
     - mockexec
     - mockhttp
     - MYDIR
+    - nazd
```
Comment (vhvb1989) [inline .vscode/cspell.global.yaml:99]: Use this PR instead: https://github.com/Azure/azure-dev/pull/4191

`nazd` should not be here. It is an issue of cspell not recognizing string scape `\n`

**Review by vhvb1989 — APPROVED** [review]

**Comment by vhvb1989** [comment]
Please don't update the dictionaries, as there is an issue with cspell. Let's use https://github.com/Azure/azure-dev/pull/4191

File: .vscode/cspell.global.yaml (hunk 1 above)
Comment (jongio) [reply .vscode/cspell.global.yaml]: Thx, i removed

**Comment by jongio** [comment]
Let's wait on validation from @rick-roche and/or @johnnyreilly before merging.

**Comment by johnnyreilly** [comment]
Oh wow - thanks for taking a look at this! Apropos of nothing, I'm indebted to you elsewhere @jongio - you power the Auth that I use to ship my Azure Static Web Apps backed blog and even get a shout out in the [README](https://github.com/johnnyreilly/blog.johnnyreilly.com?tab=readme-ov-file#auth) 😅

I'm away on holiday right now but will be back in front of a computer next week - I can test then! From the looks of the PR we can just use:

```bash
curl -fsSL https://azuresdkreleasepreview.blob.core.windows.net/azd/standalone/pr/4190/uninstall-azd.sh | bash;
curl -fsSL https://azuresdkreleasepreview.blob.core.windows.net/azd/standalone/pr/4190/install-azd.sh | bash -s -- --base-url https://azuresdkreleasepreview.blob.core.windows.net/azd/standalone/pr/4190 --version '' --verbose --skip-verify
```

To test the PR build?

**Comment by jongio** [comment]
> To test the PR build?

Yes, thanks


That early OIDC work with scripts turned into `azd pipeline config`, hopefully you don't need to use those scripts anymore.  LMK if you do and we can see about getting azd to support whatever scenario you need.

**Comment by johnnyreilly** [comment]
> That early OIDC work with scripts turned into azd pipeline config, hopefully you don't need to use those scripts anymore. 

I set my blogs CI/CD up a long time ago - possibly before azd existed and it has worked like a champ! That said, I'm planning to migrate to use azd instead - essentially I want to use the Static Web App equivalent of the following:

https://johnnyreilly.com/using-azd-for-faster-incremental-azure-container-app-deployments-in-azure-devops

> LMK if you do and we can see about getting azd to support whatever scenario you need.

I surely will!

**Comment by jongio** [comment]
> > That early OIDC work with scripts turned into azd pipeline config, hopefully you don't need to use those scripts anymore.
> 
> I set my blogs CI/CD up a long time ago - possibly before azd existed and it has worked like a champ! That said, I'm planning to migrate to use azd instead - essentially I want to use the Static Web App equivalent of the following:
> 
> https://johnnyreilly.com/using-azd-for-faster-incremental-azure-container-app-deployments-in-azure-devops
> 
> > LMK if you do and we can see about getting azd to support whatever scenario you need.
> 
> I surely will!

Great.  I appreciate you being an azd adovcate and the feedback really helps.

**Comment by rick-roche** [comment]
Hi @jongio , this is fantastic - thank you! This adds what I was hoping for and works as expected off the branch.

**Pull request merged**